### PR TITLE
Simplify 'elapsed time' method, fix initial expected time calculation and error calculation when application is late

### PR DESCRIPTION
**Pull request by betobetodev**

### CODE REVIEW

File: monitor/plugins/spark_progress.py [hunk 1]
```diff
@@ -88,37 +89,32 @@ def _publish_measurement(self, job_request):
...
 
 
     def _get_elapsed_time(self, gmt_timestamp):
-        try:
-            local_tz = tzlocal.get_localzone()
-
-        except Exception as e:
-            local_tz = "America/Recife"
-            local_tz = pytz.timezone(local_tz)
+        submission_date = datetime.strptime(gmt_timestamp,
```
Comment (armstrongmsg) [inline monitor/plugins/spark_progress.py]: Does this change resolve the problem with different timezones?

File: monitor/plugins/spark_progress.py [hunk 2]
```diff
@@ -88,37 +89,32 @@ def _publish_measurement(self, job_request):
                 ref_value = (elapsed_time / self.job_expected_time)
 
                 # Error
-                error = job_progress - ref_value
-                
+                if self.job_expected_time < 0.0:
```
Comment (armstrongmsg) [inline monitor/plugins/spark_progress.py:92]: I think you should treat this case different. What if remaining time is zero? self.job_expected_time is calculated as zero and then divide by zero exception is raised in line 89.

**Review by armstrongmsg — CHANGES_REQUESTED** [review]

File: monitor/plugins/spark_progress.py (hunk 2 above)
Comment (betobetodev) [reply monitor/plugins/spark_progress.py:92]: @armstrongmsg What is your suggestion?

File: monitor/plugins/spark_progress.py (hunk 1 above)
Comment (betobetodev) [reply monitor/plugins/spark_progress.py]: The problem with different timezones is solved when the two hosts (WP3 infrastructure and cluster) is synchronized. These sync must be done in deploy stage.

File: monitor/plugins/spark_progress.py (hunk 2 above)
Comment (armstrongmsg) [reply monitor/plugins/spark_progress.py:92]: line 75
if remaining_time <= 0:
      job_expected_time = 0
else:
      ...

line 89
if job_expected_time <= 0:
      error = 1.0
else:
       ref_value = ...
       error = job_progress ...

**Pull request closed without merging**